src/routes/users/postUsers.ts

```diff
@@ -0,0 +1,54 @@
+import express, { Request, Response, NextFunction } from 'express';
+import db from '../../knex';
+import bcrypt from 'bcrypt';
+import jwt from 'jsonwebtoken';
+import { validate, setLocales, en } from 'robust-validator';
+
+const router = express.Router();
+const JWT_SECRET = process.env.JWT_SECRET || 'key';
+const saltRounds = 10;
+
+// Validation setup
+setLocales(en);
+
+const createUserValidation = {
+  email: 'required|email',
+  password: 'required|min:6|max:50',
+  full_name: 'required|min:3|max:100', 
+};
+
+router.post('/', async (req: Request, res: Response, next: NextFunction) => {
+  const { email, password, full_name } = req.body; 
+
+  // Validation
+  const result = await validate({ email, password, full_name }, createUserValidation);
+
+  if (Object.keys(result.errors).length > 0) {
```

Comment:
```suggestion
  if (result. isInvalid) {
```

https://validator.axe-api.com/getting-started.html#usage